bubuku/features/rebalance_by_size.py

```diff
@@ -0,0 +1,116 @@
+import json
+import logging
+
+from kazoo.exceptions import NodeExistsError
+
+from bubuku.broker import BrokerManager
+from bubuku.controller import Check, Change
+from bubuku.utils import CmdHelper
+from bubuku.zookeeper import Exhibitor
+
+_LOG = logging.getLogger('bubuku.features.rebalance_by_size')
+
+
+class RebalanceBySizeChange(Change):
+    def __init__(self, zk: Exhibitor):
+        self.zk = zk
+
+    def get_name(self):
+        return 'rebalance_by_size'
+
+    def can_run(self, current_actions):
+        raise NotImplementedError('Not implemented')  # todo: not implemented yet
+
+    def run(self, current_actions):
+        raise NotImplementedError('Not implemented')  # todo: not implemented yet
+
+    def can_run_at_exit(self):
+        raise NotImplementedError('Not implemented')  # todo: not implemented yet
+
+
+class RebalanceBySize(Check):
+    def __init__(self, zk: Exhibitor, broker: BrokerManager):
+        super().__init__(check_interval_s=600)
+        self.zk = zk
+        self.broker = broker
+
+    def check(self):
+        if self.broker.is_running_and_registered() and self.__is_data_imbalanced():
+            _LOG.info("Starting rebalance by size")
+            return RebalanceBySizeChange(self.zk)
+        return None
+
+    def __is_data_imbalanced(self) -> bool:
+        return False  # todo: not implemented yet
+
+
+class GenerateDataSizeStatistics(Check):
+    def __init__(self, zk: Exhibitor, broker: BrokerManager, cmd_helper: CmdHelper, kafka_log_dirs):
+        super().__init__(check_interval_s=1800)
+        self.zk = zk
+        self.broker = broker
+        self.cmd_helper = cmd_helper
+        self.kafka_log_dirs = kafka_log_dirs
```

Comment:
Why not? I don't really see a difference here if to store it as a string or as an array.